### ISSUE TITLE
(aws-s3): (Buckets set up to save Access Logs to the same bucket are not saving logs)

### ISSUE THREAD

**Issue by biffgaut** [aws/aws-cdk#23516]
### Describe the bug

Setting serverAccessLogsPrefix in Bucket.props without setting  serverAccessLogsBucket is supposed to set up a bucket that saves logs back to the content bucket. The current implementation sets up the bucket, but doesn't grant any privileges to logDeliveryWrite, so no logs are actually created.

This may require more discussion as to whether the expected behavior (a bucket logging to itself) is an anti-pattern that should not be allowed implicitly based on input properties due to it's impact on storage.

### Expected Behavior

Access logs for Bucket A should be stored directly on Bucket A.

### Current Behavior

The stack launches without incident or warning, the bucket is set up to log to itself, but the permissions are not configured so no logs are created (full CFN snippet is included below).

### Reproduction Steps

```
export class SsebugStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // Self Logging - no encryption
    new s3.Bucket(this, 'selfNoEnc', {
      bucketName: 'bgc-source-selflogging-noencryption',
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
      serverAccessLogsPrefix: 'logs',
      encryption: BucketEncryption.UNENCRYPTED
    });
  }
}
```

### Possible Solution

Grant the proper permissions

### Additional Information/Context

It's questionable whether this "bug" should be fixed. Storing logs on the original content bucket leads to generating logs from logs - theoretically an infinite loop. It is not recommended in the [docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html), and buckets configured in this way get a warning int the console:

![image](https://user-images.githubusercontent.com/78155736/210104953-4de8ed5a-fd5d-494e-acdc-ffba27c53c76.png)

While the internal comments in the code show that self logging is the result of this configuration, the [docs](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html) on the Web never mention the possibility. 

### CDK CLI Version

2.55.1

### Framework Version

2.58.0

### Node.js Version

16.17.0

### OS

macOS Monterey 12.6.1

### Language

Typescript

### Other information

This is the cloudformation template snippet created by the code above. Note there is no privilege to logDeliveryWrite in the ACLs nor the Bucket Policy:

```
  "selfNoEnc79E1ADB1": {
   "Type": "AWS::S3::Bucket",
   "Properties": {
    "BucketName": "bgc-source-selflogging-noencryption",
    "LoggingConfiguration": {
     "LogFilePrefix": "logs"
    },
    "Tags": [
     {
      "Key": "aws-cdk:auto-delete-objects",
      "Value": "true"
     }
    ]
   },
   "UpdateReplacePolicy": "Delete",
   "DeletionPolicy": "Delete",
   "Metadata": {
    "aws:cdk:path": "SsebugStack/selfNoEnc/Resource"
   }
  },
  "selfNoEncPolicy05F5B2D9": {
   "Type": "AWS::S3::BucketPolicy",
   "Properties": {
    "Bucket": {
     "Ref": "selfNoEnc79E1ADB1"
    },
    "PolicyDocument": {
     "Statement": [
      {
       "Action": [
        "s3:DeleteObject*",
        "s3:GetBucket*",
        "s3:List*"
       ],
       "Effect": "Allow",
       "Principal": {
        "AWS": {
         "Fn::GetAtt": [
          "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
          "Arn"
         ]
        }
       },
       "Resource": [
        {
         "Fn::GetAtt": [
          "selfNoEnc79E1ADB1",
          "Arn"
         ]
        },
        {
         "Fn::Join": [
          "",
          [
           {
            "Fn::GetAtt": [
             "selfNoEnc79E1ADB1",
             "Arn"
            ]
           },
           "/*"
          ]
         ]
        }
       ]
      }
     ],
     "Version": "2012-10-17"
    }
   },
   "Metadata": {
    "aws:cdk:path": "SsebugStack/selfNoEnc/Policy/Resource"
   }
  },
```

**Comment by kylelaker** [comment]
This was discussed previously in https://github.com/aws/aws-cdk/pull/20898#discussion_r913713551 which was closed without being merged. In #23386, the functionality when the `@aws-cdk/aws-s3:serverAccessLogsUseBucketPolicy` feature flag is enabled _should_ apply the correct bucket policy for writing these logs. I think the same condition that now adds the bucket policy also adds the ACL.

I don't think that #23386 has been included in a release yet.

**Comment by peterwoodworth** [comment]
Thanks for providing this context @kylelaker, it seems to fully address this issue. You're right that this isn't included in our latest release yet, it should be out within a week or so 🙂 

@biffgaut the intended functionality is described in our docs [here](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html#serveraccesslogsprefix)

**Comment by biffgaut** [comment]
Probably ought to make sure it is discussed here in the docs as well:

![image](https://user-images.githubusercontent.com/78155736/210116815-555c958e-c47d-4444-96e2-6eba68e59e9f.png)

For such a simple property many people may look no further than this table.

**Comment by peterwoodworth** [comment]
Is this issue resolved?

**Comment by biffgaut** [comment]
@peterwoodworth This [PR](https://github.com/aws/aws-cdk/pull/23386) should have addressed it, it was released in 2.59.0 on 1/3. It may require setting a feature flag to access the functionality.
